### ISSUE TITLE
As a user I would to understand on what constitutes as a valid block so that I can create my own block validation implementation.

### ISSUE THREAD

**Issue by thec00n** [omisego/elixir-omg#1183]
### Description 
Currently there is no specification on what constitutes as valid block. Block validation is a critical function of the network and validation should not strictly be limited to the watcher but be possible in any implementation that a user chooses. Similar to transaction validation https://github.com/omisego/elixir-omg/blob/master/docs/transaction_validation.md there should be a spec for block validation.

**Comment by InoMurko** [comment]
@achiurizo ex_plasma?

**Comment by achiurizo** [comment]
yep, it's certainly a good place to put this kind of knowledge/behavior in [ex_plasma](https://github.com/omisego/ex_plasma). It can already handle crafting transactions, utxos(input/output), and block formations.

Idea I'm playing around is if it makes sense to use the [ecto changeset](https://hexdocs.pm/ecto/Ecto.Changeset.html) functionality to help implement validation. It also would pave the way towards using ecto as our storage driver.

**Comment by pdobacz** [comment]
> Idea I'm playing around is if it makes sense to use the [ecto changeset](https://hexdocs.pm/ecto/Ecto.Changeset.html)

this in particular is on my mind since long, never got to it :cry:
